src/System.CommandLine.StaticCompletions/shells/BashShellProvider.cs

```diff
@@ -0,0 +1,222 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.CommandLine.StaticCompletions.Shells;
+
+using System.CodeDom.Compiler;
+using System.CommandLine;
+using System.CommandLine.Completions;
+using System.IO;
+
+public class BashShellProvider : IShellProvider
+{
+    public string ArgumentName => "bash";
+
+    public string Extension => "sh";
+
+    public string GenerateCompletions(CliCommand command)
+    {
+        var initialFunctionName = command.FunctionName().MakeSafeFunctionName();
+        return
+            $"""
+            #! /bin/bash
```

Comment:
A nit: the proper syntax is without the space and this one is recommended to make scripts portable across the distros:

```suggestion
            #!/usr/bin/env bash
```

Reply:
Thanks - I'll apply that locally because it'll also cause changes in the snapshots that I'll need to cover at the same time. Keep the correctness feedback coming - over time the generator can get incrementally better!

Reply:
I have a little hesitation with respect to the security of `#!/usr/bin/env bash`. I've normally used `#!/usr/bin/bash` in the past, and that pinpoints the exact executable to use rather than grabbing the one specified in $PATH, which is more far easier to tamper with.

Reply:
/usr/bin/bash is not guaranteed on all platforms and it is recommended to use environment. This is how we are doing it in dotnet repos for several years. Compromised environment is not going to take us far.

No need to delete #! either, it helps testing the script in isolation. I always put it in interpreter as well as include scripts. Does not hurt. CRLF issue is also not limited to the interpreter line.